### ISSUE TITLE
Pathfinding is inaccurate and quite dumb

### ISSUE THREAD

**Issue by n1ght4ngel19** [n1ght4ngel19/gruntz-unityverse#18]
During pathfinding, Gruntz do start moving towards their target, but there are problems. First of all, mostly when they have to navigate a path with obstacles, they might get stuck on them. Second, when only a "wall" separates Gruntz from their goal, but there are ways to reach that goal by going around, they might (and mostly do) want to take a direct path, through the solid tile, which, of course, isn't possible, so they get stuck here too. And last, but not least, when Gruntz do find their path and do not get stuck on anything, they might not even properly step on their goal, but beside it on another tile, or in between tiles.

TLDR; There is a basic and dumb pathfinding implemented, but there is some work to do on that.

**Comment by n1ght4ngel19** [comment]
An issue might be that gCost, hCost, and previous aren't cleared after a Node isn't in use

**Comment by n1ght4ngel19** [comment]
Or path needs to be calculated with weights (diagonals vs straight lines)

**Comment by n1ght4ngel19** [comment]
Or the positionsing, indexing, or whatever of Nodes just isn't good

**Comment by n1ght4ngel19** [comment]
Or it's a mix of these, none of these, all of these, or it's just bad all the way

**Comment by n1ght4ngel19** [comment]
Resolved it in this #63 commmit
